source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -612,6 +619,46 @@ def _getAvailableAddonsInBG(self):
 		core.callLater(delay=0, callable=self.detailsVM.updated.notify, addonDetailsVM=self.detailsVM)
 		log.debug("completed refresh")
 
+	def _cancelDownloadForAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		try:
+			addonDataManager._downloadsPendingCompletion.remove(listItemVM)
+		except KeyError:
+			log.debug("Download already completed")
+		else:
+			futuresCopy = self._downloader._pending.copy()
+			for future, addon in futuresCopy.items():
+				if listItemVM == addon[0] and not future.cancel():
+					self._downloader.progress.pop(addon[0], None)
+		finally:
+			listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)
+
+	def _cancelPendingInstallForAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		pendingInstallCopy = addonDataManager._downloadsPendingInstall.copy()
+		for addonData, fileDownloaded in pendingInstallCopy:
+			if addonData == listItemVM:
+				addonDataManager._downloadsPendingInstall.remove((addonData, fileDownloaded))
+				listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)

```

Comment:
I would suggest moving all of the status updates to the end of cancelInstallForAddon

```suggestion
```